AdyenCard/Components/GiftCardComponent/GiftCardComponent.swift

```diff
@@ -287,8 +286,12 @@ public final class GiftCardComponent: PartialPaymentComponent,
 
 extension Result {
 
+    func handle(_ success: (Success) -> Void, _ failure: (Failure) -> Void) {
```

Comment:
maybe better to have parameters of this function visible, because this is a bit confusing
```
orderResult.handle(submit(order:)) { delegate?.didFail(with: $0, from: self) }
```